http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/service/DiskServiceInterp.scala

```diff
@@ -318,6 +318,87 @@ class DiskServiceInterp[F[_]: Parallel](config: PersistentDiskConfig,
 
     } yield ()
 
+  def deleteAllOrphanedDisks(userInfo: UserInfo,
+                             cloudContext: CloudContext.Gcp,
+                             runtimeDiskIds: Vector[DiskId],
+                             appDisksNames: Vector[Option[DiskName]]
+  )(implicit
+    as: Ask[F, AppContext]
+  ): F[Unit] =
+    for {
+      ctx <- as.ask
+      // Get a list of disks that are not attached to any runtime or app
+      disks <- listDisks(
+        userInfo,
+        Some(cloudContext),
+        Map.empty
+      )
+
+      orphanedDisks = disks.filterNot(disk =>
+        runtimeDiskIds.contains(disk.id) | appDisksNames.contains(Some(disk.name))
```

Comment:
Super nit -- was there a choice to use `|` instead of `||`? I think the difference is that `||` is short circuiting (I had to google it 😄 https://stackoverflow.com/questions/25067702/scala-booleans-vs-vs)

Reply:
Oh no choice on my part, but short circuiting makes sense, I will add it :)